### ISSUE TITLE
Nucache lock with SqlMainDomLock in place

### ISSUE THREAD

**Issue by rickbracedigital** [umbraco/Umbraco-CMS#9517]
I'm seeing the following error in the logs  a lot which causes the website to hang and ultimately go down:

![image](https://user-images.githubusercontent.com/17140151/101534982-dd400900-398f-11eb-8e39-e5745ab1f75d.png)
I have flexible load balancing implemented and as per the load balancing docs, I have the appSetting <add key="Umbraco.Core.MainDom.Lock" value="SqlMainDomLock" /> added in the "Master" and "Replica" app services. This is added via the Azure portal.

The website goes down almost every day (at totally different parts of the day) and I'm seeing this occur when the website scales (I have auto-scaling configured for the "Replica" app service).

## Umbraco version

I am seeing this issue on Umbraco version: 8.9.0

Reproduction
------------

Follow the load balancing docs to the letter for Umbraco 8.9.0 including adding the Umbraco.Core.MainDom.Lock setting for Umbraco 8.9.0. Configure auto-scaling for the "Replica" app service only.

### Bug summary

![image](https://user-images.githubusercontent.com/17140151/101535081-03fe3f80-3990-11eb-971e-08f6a92af3e7.png)

Umbraco 8.9.0

This occurs when the website auto-scales - as soon as I get the email from Azure to say that it has auto-scaled I get a Pingdom (monitoring client) alert to say that the site is down. The log files are littered with the Nucache error message above.

Thanks,
Rick

**Comment by nul800sebastiaan** [comment]
Thanks @rickbracedigital, we do see this from time to time and it comes down to using the correct configuration as listed here: https://our.umbraco.com/Documentation/Getting-Started/Setup/Server-Setup/azure-web-apps#recommended-configuration


For more info and a loooong thread, make sure to give this issue a read https://github.com/umbraco/Umbraco-CMS/issues/5035

**Comment by rickbracedigital** [comment]
Hi Sebastiaan,

I’m still seeing the issue despite following the instructions to the letter. Instead of closing the issue, please can you advise?

**Comment by nul800sebastiaan** [comment]
That's very unfortunate! The only thing I can do is to ask you to try this out with a vanilla install and see if you can reproduce it, if you can then we'd need to know details of the exact setup so we can reproduce and debug. It's always really hard with seemingly random issues that only seem to occur for a single user. I always check if there's other people reporting this error in the latest versions so I'll add here: `The process cannot access the file NuCache.Content.db because it is being used by another process` so text searches will turn up this issue in the future. 

I would recommend you try to repro and keep monitoring and diving into logs to see if it always seems to happen near some other event.

**Comment by m4nickroll** [comment]
@nul800sebastiaan - there seems to be some mixed messages around the WEBSITE_DISABLE_OVERLAPPED_RECYCLING flag when researching this issue. This documentation - https://our.umbraco.com/Documentation/Getting-Started/Setup/Server-Setup/Load-Balancing/azure-web-apps seems to suggest that it was only needed for v8.6.0 - v8.6.3 but I have seen anecdotally that this has help some sites running 8.6.4+. Is there a definitive answer on this?

**Comment by Shazwazza** [comment]
The documentation is correct. `WEBSITE_DISABLE_OVERLAPPED_RECYCLING` is a work around for previous behavior until SqlMainDomLock is in place.

**Comment by m4nickroll** [comment]
Thanks for confirming

**Comment by rickbracedigital** [comment]
Hi @nul800sebastiaan - I've turned auto-scaling off and I'm still seeing the website go down with the above error so seems unrelated to scaling. As I say, I've followed the documentation to the letter but still happening. There's nothing in the Umbraco logs (literally nothing) at the time when the site goes down and using Application Insights, there's nothing of note that appears to cause it either.
